scripts/generate-sdk/languages/java.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+# This script clones the SDK repo and updates it with the generated API modules
+# Pre-requisites: Java
+set -eo pipefail
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+SDK_REPO_LOCAL_PATH="${ROOT_DIR}/sdk-repo-updated"
+
+SERVICES_FOLDER="${SDK_REPO_LOCAL_PATH}/services"
+
+GENERATOR_LOG_LEVEL="error" # Must be a Java log level (error, warn, info...)
+
+INCLUDE_SERVICES=("resourcemanager")
+
+generate_java_sdk() {
+    # Required parameters
+    local GENERATOR_JAR_PATH=$1
+    local GIT_HOST=$2
+    local GIT_USER_ID=$3
+
+    # Optional parameters
+    local GIT_REPO_ID=$4
+    local SDK_REPO_URL=$5
+    local SDK_BRANCH=$6
+
+    # Check required parameters
+    if [[ -z ${GIT_HOST} ]]; then
+        echo "! GIT_HOST not specified."
+        exit 1
+    fi
+
+    if [[ -z ${GIT_USER_ID} ]]; then
+        echo "! GIT_USER_ID id not specified."
+        exit 1
+    fi
+
+    # Check optional parameters and set defaults if not provided
+    if [[ -z ${GIT_REPO_ID} ]]; then
+        echo "GIT_REPO_ID not specified, default will be used."
+        GIT_REPO_ID="stackit-sdk-java"
+    fi
+
+    if [[ -z ${SDK_REPO_URL} ]]; then
+        echo "SDK_REPO_URL not specified, default will be used."
+        SDK_REPO_URL="https://github.com/stackitcloud/stackit-sdk-java.git"
+    fi
+
+    # Prepare folders
+    if [[ ! -d $SERVICES_FOLDER ]]; then
+        mkdir -p "$SERVICES_FOLDER"
+    fi
+
+    # Clone SDK repo
+    if [ -d ${SDK_REPO_LOCAL_PATH} ]; then
+        echo "Old SDK repo clone was found, it will be removed."
+        rm -rf ${SDK_REPO_LOCAL_PATH}
+    fi
+    git clone --quiet -b ${SDK_BRANCH} ${SDK_REPO_URL} ${SDK_REPO_LOCAL_PATH}
+
+    # Backup of the current state of the SDK services dir (services/)
+    sdk_services_backup_dir=$(mktemp -d)
+    if [[ ! ${sdk_services_backup_dir} || -d {sdk_services_backup_dir} ]]; then
+        echo "! Unable to create temporary directory"
+        exit 1
+    fi
+    cleanup() {
+        rm -rf ${sdk_services_backup_dir}
+    }
+    cp -a "${SERVICES_FOLDER}/." ${sdk_services_backup_dir}
+
+    # Cleanup after we are done
+    trap cleanup EXIT
+
+    # Remove old contents of services dir (services/)
+    rm -rf ${SERVICES_FOLDER}
+
+    # Generate SDK for each service
+    for service_json in ${ROOT_DIR}/oas/*.json; do
+        service="${service_json##*/}"
+        service="${service%.json}"
+
+        # Remove invalid characters to ensure a valid Go pkg name
```

Comment:
```suggestion
        # Remove invalid characters to ensure a valid java package name
```

Please check what's allowed in java package names and adjust this as needed.

Reply:
The oracle docs you linked say to use the underscore in case the package name conflicts with reserved java keywords. Since I wouldn't expect this here, I would also remove underscores from the package name :)

Reply:
Yes, I also don't think that this should happen and changed it. I left the if clause, that the servicename will a underscore as prefix if it starts with a number. I don't think that this will be the case, but you never know 😄